lib/MusicBrainz/Server/View/Node.pm

```diff
@@ -38,20 +39,28 @@ sub process {
             sessionid => scalar($c->sessionid),
             session => $c->session,
             flash => $c->flash,
-        }});
+        },
+    };
 
     my $uri = URI->new;
     $uri->scheme('http');
     $uri->host(DBDefs->RENDERER_HOST || '127.0.0.1');
     $uri->port(DBDefs->RENDERER_PORT);
     $uri->path($c->req->path);
+    $uri->query_param_append('token', generate_token());
+    $uri->query_param_append('user', $c->user->name) if $c->user_exists;
```

Comment:
The token is really enough, but adding the user too should make it impossible to get a page requested by another user (disaster scenario being that it contains private data).